sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/rx/NonStreamingOrderByQueryVectorSearchTest.java

```diff
@@ -0,0 +1,295 @@
+import com.azure.cosmos.models.CosmosVectorIndexType;
+import com.azure.cosmos.models.IncludedPath;
+import com.azure.cosmos.models.IndexingMode;
+import com.azure.cosmos.models.IndexingPolicy;
+import com.azure.cosmos.models.PartitionKeyDefinition;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import reactor.core.publisher.Flux;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+
+import static com.azure.cosmos.rx.TestSuiteBase.createDatabase;
+import static com.azure.cosmos.rx.TestSuiteBase.safeClose;
+import static com.azure.cosmos.rx.TestSuiteBase.safeDeleteDatabase;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class NonStreamingOrderByQueryVectorSearchTest {
+    protected static final int TIMEOUT = 30000;
+    protected static final int SETUP_TIMEOUT = 20000;
+    protected static final int SHUTDOWN_TIMEOUT = 20000;
+
+    protected static Logger logger = LoggerFactory.getLogger(NonStreamingOrderByQueryVectorSearchTest.class.getSimpleName());
+
+    private final String databaseId = CosmosDatabaseForTest.generateId();
+    private final String flatContainerId = "flat_" + UUID.randomUUID();
+    private final String quantizedContainerId = "quantized_" + UUID.randomUUID();
+    private final String largeDataContainerId = "large_data_" + UUID.randomUUID();
+    private CosmosAsyncClient client;
+    private CosmosAsyncDatabase database;
+    private CosmosAsyncContainer flatIndexContainer;
+    private CosmosAsyncContainer quantizedIndexContainer;
+    private CosmosAsyncContainer largeDataContainer;
+
+    @BeforeClass(groups = {"query"}, timeOut = SETUP_TIMEOUT)
+    public void before_NonStreamingOrderByQueryVectorSearchTest() {
+        // set up the client
+        client = new CosmosClientBuilder()
+            .endpoint(TestConfigurations.HOST)
+            .key(TestConfigurations.MASTER_KEY)
+            .directMode(DirectConnectionConfig.getDefaultConfig())
+            .consistencyLevel(ConsistencyLevel.SESSION)
+            .contentResponseOnWriteEnabled(true)
+            .buildAsyncClient();
+
+        database = createDatabase(client, databaseId);
+
+        PartitionKeyDefinition partitionKeyDef = new PartitionKeyDefinition();
+        ArrayList<String> paths = new ArrayList<String>();
+        paths.add("/pk");
+        partitionKeyDef.setPaths(paths);
+
+        CosmosContainerProperties containerProperties = new CosmosContainerProperties(flatContainerId, partitionKeyDef);
+        containerProperties.setIndexingPolicy(populateIndexingPolicy(CosmosVectorIndexType.FLAT));
+        containerProperties.setVectorEmbeddingPolicy(populateVectorEmbeddingPolicy(128L));
+        database.createContainer(containerProperties).block();
```

Comment:
Can we also add tests with container with multi-partitions?

Reply:
Added tests with multi-partitions as well